sql/004_search.sql

```diff
@@ -637,33 +637,33 @@ CREATE OR REPLACE FUNCTION search_tohash(jsonb) RETURNS jsonb AS $$
     SELECT $1 - '{token,limit,context,includes,excludes}'::text[];
 $$ LANGUAGE SQL IMMUTABLE PARALLEL SAFE;
 
-CREATE OR REPLACE FUNCTION search_hash(jsonb) RETURNS text AS $$
-    SELECT md5(search_tohash($1)::text);
+CREATE OR REPLACE FUNCTION search_hash(jsonb, jsonb) RETURNS text AS $$
+    SELECT md5(concat($1::text, $2::text));
 $$ LANGUAGE SQL IMMUTABLE PARALLEL SAFE;
 
-
 CREATE TABLE IF NOT EXISTS searches(
-    hash text GENERATED ALWAYS AS (search_hash(search)) STORED PRIMARY KEY,
+    hash text GENERATED ALWAYS AS (search_hash(search, metadata)) STORED PRIMARY KEY,
     search jsonb NOT NULL,
     _where text,
     orderby text,
     lastused timestamptz DEFAULT now(),
     usecount bigint DEFAULT 0,
     statslastupdated timestamptz,
     estimated_count bigint,
-    total_count bigint
+    total_count bigint,
+    metadata jsonb DEFAULT '{}'::jsonb
```

Comment:
maybe we don't have to set a default here (but then we should have a default in the `search_hash` function)

Reply:
I think it's good to have a default and probably set it to not null.

That way we can merge update the metadata by using "UPDATE searches SET metadata = metadata || '{"newfield": 1}':::jsonb without having to do any coalesce or anything to deal with nulls.

Reply:
basically I set metadata to default to `{}` to make sure `search_hash` received two valid json object 🤷‍♂️